launch_testing/launch_testing/__init__.py

```diff
@@ -70,8 +73,11 @@ def on_stdout_lines(self, lines):
 
         for line in lines.splitlines():
             # Filter out stdout that comes from underlying DDS implementation
+            # Note: we do not currently support matching filters across multiple stdout lines.
             if any(line.startswith(prefix) for prefix in self.filtered_prefixes):
                 continue
+            if any(re.match(pattern, line) for pattern in self.filtered_patterns):
```

Comment:
I was first worried that using the same pattern without compiling it repeatedly would be slower. But it turns out modern Python already do a great job compiling the pattern and caching it for future use. So all good here :smiley: